### ISSUE TITLE
Android Studio 2.2  Preview timed out while rendering the layout.

### ISSUE THREAD

**Issue by recwert** [brianwernick/ExoMedia#341]
When include the exoMedia to xml layout, the android studio 2.2 preview throw exception
java.util.concurrent.TimeoutException: Preview timed out while rendering the layout.
This typically happens when there is an infinite loop or unbounded recursion in one of the custom views.
	at java.lang.Object.wait(Native Method)
	at java.lang.Object.wait(Object.java:502)
	at com.google.android.exoplayer.ExoPlayerImplInternal.release(ExoPlayerImplInternal.java:188)
	at com.google.android.exoplayer.ExoPlayerImpl.release(ExoPlayerImpl.java:156)
	at com.devbrackets.android.exomedia.exoplayer.EMExoPlayer.release(EMExoPlayer.java:275)
	at com.devbrackets.android.exomedia.EMVideoView.release(EMVideoView.java:359)
	at com.devbrackets.android.exomedia.EMVideoView.onDetachedFromWindow(EMVideoView.java:142)
	at android.view.View.dispatchDetachedFromWindow(View.java:15443)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3224)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3216)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3216)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3216)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3216)
	at android.view.ViewGroup.dispatchDetachedFromWindow(ViewGroup.java:3216)
	at android.view.AttachInfo_Accessor.detachFromWindow(AttachInfo_Accessor.java:51)
	at com.android.layoutlib.bridge.impl.RenderSessionImpl.dispose(RenderSessionImpl.java:1469)
	at com.android.layoutlib.bridge.BridgeRenderSession.dispose(BridgeRenderSession.java:203)
	at com.android.tools.idea.rendering.RenderTask$$Lambda$176/1729223569.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by hibob224** [comment]
Had the same issue today. Was fixed by invalidating AS cache

**Comment by mucuspigboy** [comment]
Hi, I'm getting this problem now. Really strange, as I'm doing a Hello World. (Sorry, am new to this - first time using android studio). So, I don't know what the problem is.

Hibob224 - "invalidating AS cache"? Put another way, are you saying reinstall android studio?

Thanks,
p

**Comment by Arlem** [comment]
No, it's an option under File > Invalidate Caches / Restart...

**Comment by mucuspigboy** [comment]
Thanks, dude. I'll give it a try.

Before your reply, however, I did reinstall. Although I still get the same error, at least this time I reached "gradle finished building (in 7m 38s)". I think that's way too long for a "Hello World" app, so something is probably wrong with my system.

I got other stuff to do, so it's probably going to take me awhile to get back on this, but I will post any developments in case anyone is interested.
